packages/dataviews/src/field-types/telephone.tsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import type {
+	DataViewRenderFieldProps,
+	SortDirection,
+	NormalizedField,
+	FieldTypeDefinition,
+} from '../types';
+import { renderFromElements } from '../utils';
+import {
+	OPERATOR_IS,
+	OPERATOR_IS_ALL,
+	OPERATOR_IS_NOT_ALL,
+	OPERATOR_IS_ANY,
+	OPERATOR_IS_NONE,
+	OPERATOR_IS_NOT,
+	OPERATOR_CONTAINS,
+	OPERATOR_NOT_CONTAINS,
+	OPERATOR_STARTS_WITH,
+} from '../constants';
+
+function sort( valueA: any, valueB: any, direction: SortDirection ) {
+	return direction === 'asc'
+		? valueA.localeCompare( valueB )
+		: valueB.localeCompare( valueA );
+}
+
+// Basic international phone number validation
+// Supports formats like: +1234567890, (123) 456-7890, 123-456-7890, 123.456.7890, etc.
+const phoneRegex =
+	/^[\+]?[(]?[\+]?\d{0,3}[)]?[-\s\.]?\d{0,3}[-\s\.]?\d{3,4}[-\s\.]?\d{3,4}$/;
+
+export default {
+	sort,
+	isValid: {
+		custom: ( item: any, field: NormalizedField< any > ) => {
+			const value = field.getValue( { item } );
+
+			if (
+				! [ undefined, '', null ].includes( value ) &&
+				! phoneRegex.test( value )
+			) {
+				return __( 'Value must be a valid phone number.' );
+			}
+
+			if ( field.elements ) {
+				const validValues = field.elements.map( ( f ) => f.value );
+				if ( ! validValues.includes( value ) ) {
+					return __( 'Value must be one of the elements.' );
+				}
+			}
+
+			return null;
+		},
```

Comment:
This is a good opportunity to introduce `isValid.pattern`. I consider `custom` a scape hatch for things we don't plan to absorb. It's fine to ship this PR without validation and implement `isValid.pattern` as a follow-up, if you'd rather do that (as opposed to implement `isValid.pattern` here).

Reply:
Yes, something is definitely needed here as lines are identical. I haven't had the chance to familirize myself with the details and would like to look at it in a quick follow up, to have time to think about it.